Adds documentation and improves migrations failing on timeouts while waiting for index yellow status (#130352)
elastic/kibana: docs/setup/upgrade/resolving-migration-failures.asciidoc

setting prevents upgrade migrations from starting.

[float]
==== Repeated time-out requests that eventually fail
Migrations get stuck in a loop of retry attempts waiting for index yellow status that's never reached.
In the CLONE_TEMP_TO_TARGET or CREATE_REINDEX_TEMP steps, you might see a log entry similar to:

[source,sh]
--------------------------------------------
"Action failed with [index_not_yellow_timeout] Timeout waiting for the status of the [.kibana_8.1.0_001] index to become "yellow". Retrying attempt 1 in 2 seconds."
--------------------------------------------
The process is waiting for a yellow index status. There are two known causes:

* Cluster hits the low watermark for disk usage
* Cluster has <<routing-allocation-disabled,routing allocation disabled>>

Before retrying the migration, inspect the output of the `_cluster/allocation/explain?index=${targetIndex}` API to identify why the index isn't yellow:

[source,sh]
--------------------------------------------
GET _cluster/allocation/explain
{
  "index": ".kibana_8.1.0_001",
  "shard": 0,
  "primary": true,
}
--------------------------------------------

> If the cluster exceeded the low watermark for disc usage, the output should contain a message similar to this:

Reviewer: ```suggestion
If the cluster exceeded the low watermark for disk usage, the output should contain a message similar to this:
```

[source,sh]
--------------------------------------------
"The node is above the low watermark cluster setting [cluster.routing.allocation.disk.watermark.low=85%], using more disk space than the maximum allowed [85.0%], actual free: [11.692661332965082%]"
--------------------------------------------
Refer to the {es} guide for how to {ref}/fix-common-cluster-issues.html#_error_disk_usage_exceeded_flood_stage_watermark_index_has_read_only_allow_delete_block[fix common cluster issues].

> If routing allocation is the issue, you should see an entry similar to this:

Reviewer: ```suggestion
If routing allocation is the issue, the `_cluster/allocation/explain` API will return an entry similar to this:
```
